Debian package generation and added motor project

Witherking25 opened the pull request.

Witherking25 commented:
wrong one

rofl0r commented on Makefile:
```diff
@@ -1,4 +1,4 @@
-prefix=/usr/local
+prefix=/usr
```
again, this is unneeded.

Witherking25 replied:
Oops

Witherking25 replied:
Uh. What are you looking at the branch I’m merging says /usr/local

Witherking25 replied:
This review is outdated and was already fixed

rofl0r commented on Makefile:
```diff
@@ -43,7 +43,7 @@ motor: $(OBJS)
 
 %.o: %.cc
-	$(CXX) -DSHARE_DIR=\"$(datadir)/motor\" $(INCLUDES) $(CPPFLAGS) $(CXXFLAGS) -c -o  $@ $<
+	$(CXX) -DSHARE_DIR="\"$(datadir)/motor\"" $(INCLUDES) $(CPPFLAGS) $(CXXFLAGS) -c -o  $@ $<
```
why's that necessary? it builds fine without the change here.
also this has nothing to do with the commit message

Witherking25 replied:
Defines replace text directly. It was putting /usr/share/motor in the code
Instead of the string literal “/usr/share/motor”

Witherking25 replied:
I had a bunch of commits that had multiple different things in one commit

Witherking25 replied:
It defaults to /usr/local/share so it appeared to still work but when I had changed prefix to /usr it still said /usr/local. Putting the quotes around that fixed that

rofl0r replied:
if the macro would expand to /usr/share without quote where a string is expected this would simply break the compilation. so it has nothing to do with your /usr vs /usr/local problem.

Witherking25 replied:
Well for some reason changing prefix to /usr in the makefile still put /usr/local/share/motor in the share define until I put those quotes there. Motor wasn’t finding the templates and I was trying to figure out why

Witherking25 replied:
This is a problem because I couldn’t set prefix if I wanted motor to be able to find its templates

Witherking25 replied:
I can remove them but it will break whenever I try to package motor

Witherking25 replied:
nvm im just dumb and something else fixed it. it seems to make no difference now.

Witherking25 commented:
It’s finished

rofl0r commented:
well, this PR is a mess. please make a new one that adds *only* the motor motor project in a single, clean commit, and another one for whatever this current PR does too.
i'd really recommend you to install gitk, tig, or another local git gui so you can actually see what your PRs contain and the diff hunks each commit adds.

Witherking25 commented:
Ok

Witherking25 commented:
the problem is the commit for debian packaging also edits files in the motor project so splitting them would be hard

Witherking25 commented:
```
CONFLICT (modify/delete): motor.motor deleted in HEAD and modified in df88741... added debian package generation. Version df88741... added debian package generation of motor.motor left in tree.
CONFLICT (modify/delete): Makefile.func deleted in HEAD and modified in df88741... added debian package generation. Version df88741... added debian package generation of Makefile.func left in tree.
error: could not apply df88741... added debian package generation
Resolve all conflicts manually, mark them as resolved with
"git add/rm <conflicted_files>", then run "git rebase --continue".
You can instead skip this commit: run "git rebase --skip".
To abort and get back to the state before "git rebase", run "git rebase --abort".
```

Witherking25 commented:
Try now

rofl0r commented on share/templates/terminal program/C++/Makefile.func:
```diff
@@ -1,6 +1,6 @@
 #
 # Motor IDE functional Makefile template
-# 	for terminal program in C++
+#       for terminal program in C++
```
your editor seems to automatically convert tabs into spaces. this causes a lot of unnecessary whitespace noise (check rest of the diff)

Witherking25 replied:
Vscode does that

rofl0r replied:
i don't care who does it, but i care not to merge random whitespace changes :)
it's your task to fix it up (and check whether this behaviour can be disabled in your editor)
